File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public static boolean isAnagram(String s1, String s2){
        if(s1.length() != s2.length()) return false;
        char[] a = s1.toCharArray();
        char[] b = s2.toCharArray();
        Arrays.sort(a);
        Arrays.sort(b);
        return Arrays.equals(a, b);
    }
    public List<List<String>> groupAnagrams(String[] strs) {
        List<String> list = new ArrayList<>(Arrays.asList(strs));
        List<List<String>> rs = new ArrayList<>();

        while(!list.isEmpty()){
            // Take the first element and make a new list
            String firstWord = list.remove(0);
            List<String> group = new ArrayList<>();
            group.add(firstWord);

            // Find anagrams in the remain list
            Iterator<String> iterator = list.iterator();
            while (iterator.hasNext()) {
                String s = iterator.next();
                if(isAnagram(firstWord, s)){
                    group.add(s);
                    iterator.remove();
                }
            }
            rs.add(group);
        }
        return rs;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        my_dict = dict()
        for c in s:
            if c not in my_dict:
                my_dict[c] = 1
            else:
                my_dict[c] += 1
        for c in t:
            if c in my_dict:
                my_dict[c] -= 1
                if my_dict[c] == 0:
                    my_dict.pop(c)
            else:
                return False
        return not my_dict